catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/util/impl/CachedSource.java

```diff
@@ -115,7 +115,8 @@ public boolean isAvailable() {
 
   @Override
   public boolean isAvailable(SourceMonitor callback) {
-    return isAvailable();
+    throw new UnsupportedOperationException(
```

Comment:
❗️I don't think we can simply change the behavior and throw an exception in the classes that implement `Source` since that interface is whitelisted. The current behavior should be considered a bug and we should write a ticket for it. If we throw an exception, client code will break. If we leave the existing behavior, existing client code will keep behaving like it always has.